From c13a5d8a00b6a1f1293cf7c4bc493f559b7d2591 Mon Sep 17 00:00:00 2001
From: Thierry Bultel <thierry.bultel@iot.bzh>
Date: Mon, 10 Dec 2018 16:06:23 +0100
Subject: [PATCH 3/3] increased the number of connexions to 16

---
 src/bluealsa.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bluealsa.h b/src/bluealsa.h
index b73241e..12f3c0a 100644
--- a/src/bluealsa.h
+++ b/src/bluealsa.h
@@ -31,7 +31,7 @@
 #include <bluealsa/ctl-proto.h>
 
 /* Maximal number of clients connected to the controller. */
-#define BLUEALSA_MAX_CLIENTS 7
+#define BLUEALSA_MAX_CLIENTS 16
 
 /* Indexes of special file descriptors in the poll array. */
 #define CTL_IDX_SRV 0
-- 
2.16.4

