From 14893ae87ab0b15b7e438779433c4973c797c5f5 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Wed, 29 May 2019 12:09:13 +0300
Subject: [PATCH] spa/include: install missing headers

---
 spa/include/spa/meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/spa/include/spa/meson.build b/spa/include/spa/meson.build
index c9d07659..c079a1a2 100644
--- a/spa/include/spa/meson.build
+++ b/spa/include/spa/meson.build
@@ -39,6 +39,7 @@ spa_monitor_headers = [
   'monitor/device.h',
   'monitor/monitor.h',
   'monitor/type-info.h',
+  'monitor/utils.h',
 ]
 
 install_headers(spa_monitor_headers,
@@ -50,6 +51,7 @@ spa_node_headers = [
   'node/io.h',
   'node/node.h',
   'node/type-info.h',
+  'node/utils.h',
 ]
 
 install_headers(spa_node_headers,
@@ -97,6 +99,7 @@ spa_utils_headers = [
   'utils/dict.h',
   'utils/hook.h',
   'utils/list.h',
+  'utils/result.h',
   'utils/ringbuffer.h',
   'utils/type.h',
   'utils/type-info.h',
-- 
2.20.1

