From cc5cbaddad6fe559e9e482467266fb18fb00c6a7 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Wed, 26 Jun 2019 16:02:13 +0300
Subject: [PATCH] Adapt smack rules to allow connections to pipewire

Signed-off-by: George Kiagiadakis <george.kiagiadakis@collabora.com>
---
 policy/app-rules-template.smack | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/app-rules-template.smack b/policy/app-rules-template.smack
index 910f40c..78b75de 100644
--- a/policy/app-rules-template.smack
+++ b/policy/app-rules-template.smack
@@ -4,6 +4,7 @@ System ~PKG~ rwxat
 ~APP~ System::Shared rx
 ~APP~ System::Run rwxat
 ~APP~ System::Log rwxa
+~APP~ System::Pipewire rw
 ~APP~ _ l
 ~APP~ User::Home rxl
 ~APP~ User::App-Shared rwxat
-- 
2.20.1

