From f95469247c182b3c4b527af04b1ae50658461e85 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Tue, 3 Sep 2019 16:24:49 +0300
Subject: [PATCH] Grant dbus privilege to pipewire

---
 policy/security-manager-policy-reload | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/security-manager-policy-reload b/policy/security-manager-policy-reload
index 274c49c..a883048 100755
--- a/policy/security-manager-policy-reload
+++ b/policy/security-manager-policy-reload
@@ -59,6 +59,9 @@ do
     cyad --set-policy --bucket=MANIFESTS --client="$client" --user="*" --privilege="*" --type=ALLOW
 done
 
+# PipeWire needs to get access to dbus
+cyad --set-policy --bucket=MANIFESTS --client="System::Pipewire" --user="*" --privilege="http://tizen.org/privilege/internal/dbus" --type=ALLOW
+
 # Load privilege-group mappings
 (
 echo "BEGIN;"
-- 
2.23.0.rc1

