From 3a7400da79678e749381306e33163ca2c005cb3b Mon Sep 17 00:00:00 2001
From: Thierry Bultel <thierry.bultel@iot.bzh>
Date: Mon, 3 Dec 2018 14:31:31 +0100
Subject: [PATCH 2/3] log: add calling function name

Signed-off-by: Thierry Bultel <thierry.bultel@iot.bzh>
---
 includes/bluealsa/log.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/bluealsa/log.h b/includes/bluealsa/log.h
index 39663fb..6d4cec3 100644
--- a/includes/bluealsa/log.h
+++ b/includes/bluealsa/log.h
@@ -31,7 +31,7 @@ void info(const char *format, ...) __attribute__ ((format(printf, 1, 2)));
 
 #if DEBUG
 void _debug(const char *format, ...) __attribute__ ((format(printf, 1, 2)));
-# define debug(M, ARGS ...) _debug("%s:%d: " M, __FILE__, __LINE__, ## ARGS)
+# define debug(M, ARGS ...) _debug("%s:%d:%s: " M, __FILE__, __LINE__,__func__, ## ARGS)
 #else
 # define debug(M, ARGS ...) do {} while (0)
 #endif
-- 
2.16.4

