From 2fc07fedd17650f64f9bfcdb8682b55dad75cccd Mon Sep 17 00:00:00 2001
From: Mark Farrugia <mark.farrugia@fiberdyne.com.au>
Date: Mon, 8 Apr 2019 16:38:13 +1000
Subject: [PATCH] 0001-snd-avirt-backport-kernel-4.12-api

---
 core.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/core.c b/core.c
index c2e32c5..2bb6616 100644
--- a/core.c
+++ b/core.c
@@ -219,9 +219,8 @@ int snd_avirt_stream_try_complete(struct snd_avirt_stream *stream)
 	/* Set PCM ops for the Audio Path*/
 	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, pointer);
 	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, get_time_info);
-	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, fill_silence);
-	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, copy_user);
-	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, copy_kernel);
+	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, silence);
+	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, copy);
 	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, mmap);
 	PCM_OPS_SET(pcm_ops_ap, &stream->pcm_ops, ack);
 
@@ -331,9 +330,9 @@ snd_avirt_route_endpoint_copy(struct snd_pcm_substream *substream,
 
 	switch (endpoint) {
 	case SND_AVIRT_ROUTE_SOURCE:
-		return endpoint_ap->pcm_capture_ops->copy_kernel;
+		return endpoint_ap->pcm_capture_ops->copy;
 	case SND_AVIRT_ROUTE_SINK:
-		return endpoint_ap->pcm_playback_ops->copy_kernel;
+		return endpoint_ap->pcm_playback_ops->copy;
 	}
 
 	return NULL;
-- 
2.17.1

